static/js/main.js

```diff
@@ -31,9 +31,32 @@ function findCurrentSection(htmlEl, mainEl) {
   return targetEl.id;
 }
 
+function findLIParent(node) {
+  const parent = node.parentNode;
+  
+  if(parent.nodeName === "LI") {
+    return parent;
+  }
+  
+  return findLIParent(parent);
+}
+
 function highlightNav(navEl, currentId) {
-  navEl.querySelectorAll('.current').forEach(node => node.classList.remove('current'));
-  navEl.querySelector(`[href="#${currentId}"]`).closest('p, li').classList.add('current');
+  // first remove any previously-selected highlights
+  navEl.querySelectorAll('.currentSection').forEach(node => node.classList.remove('currentSection'));
+  navEl.querySelectorAll('.currentGroup').forEach(node => node.classList.remove('currentGroup'));
+  
+  // find the current section's link in the toc
+  const sectionNode = navEl.querySelector(`[href="#${currentId}"]`).closest('p, li');
+  sectionNode.classList.add('currentSection');
+  
+  // find that section's group's li
+  let groupNode = findLIParent(sectionNode);
```

Comment:
Why not:

```
let groupNode = sectionNode.parentNode.closest('li');
```

? What am I missing?

Reply:
I've updated the pr; responses below, but I also refactored the colors in the base.scss as I kept forgetting about some of the duplicates.

> If we're going to highlight the whole group with the section, I think that the section highlight has to go all the way over to the right, rather than stopping a little short (see screenshot below).
> <img alt="Screen Shot 2021-07-18 at 12 21 48" width="363" src="https://user-images.githubusercontent.com/2257804/126074718-02c0d56a-17de-420f-bc92-391629b7a209.png">

I left a bit of space to the left and right (but now it's equal) and kicked the saturation up a notch.

> And now that the ToC has a grey background, I would experiment with having the highlights for the group and section go all the way across from the left side of the page to the right side of the ToC, rather than leaving grey on either side - and that probably means no rounded corners, either.

I'll have a look into this, but I like roundrects and I'm not sure fullwidth will quite work.

> Lastly, right now, on load, there is nothing highlighted, but once you scroll, it highlights the introduction. I think that either it should start highlighted, or it should go back to having nothing highlighted when you're scrolled back up enough (maybe only if you're scrolled up all the way). My arbitrary preference is the latter but I have no reason other than it's neat; I'm totally fine with either.

The former was easier, so that's fixed; the latter can be added later if desired.
 
> This may be a separate PR but we should consider a Footnotes section in the nav.

This requires more ast futzing code than I feel like doing today; see #59.

Reply:
> Why not:
> 
> ```
> let groupNode = sectionNode.parentNode.closest('li');
> ```
> 
> ? What am I missing?

I had to do this at work and I was _assured_ that this was not available outside jquery, so I had to recurse. Will fix!

Reply:
Well, they’re right, it wasn’t… until 2015. Also I use it just a few lines earlier :) I did think maybe I was missing something special about this case though!

Reply:
This scenario was two weeks ago - I'm just happy there was another issue, so I didn't push that code.